File: postman/sdks/Specs SDKs/Orders/python/src/orders_sdk/models/order.py
from __future__ import annotations
from enum import Enum
from pydantic import Field
from typing import Optional
from typing import Any
from .utils.base_model import BaseModel


class OrderStatus(str, Enum):
    """An enumeration representing different categories.

    :cvar PENDING: "pending"
    :vartype PENDING: str
    :cvar PROCESSING: "processing"
    :vartype PROCESSING: str
    :cvar SHIPPED: "shipped"
    :vartype SHIPPED: str
    :cvar CANCELLED: "cancelled"
    :vartype CANCELLED: str
    """

    PENDING = "pending"
    PROCESSING = "processing"
    SHIPPED = "shipped"
    CANCELLED = "cancelled"

    @staticmethod
    def list():
        """Lists all enum values.

        :return: A list of all enum values.
        :rtype: list
        """
        return list(map(lambda x: x.value, OrderStatus._member_map_.values()))


class Order(BaseModel):
    """Order

    :param id_: Unique order identifier
    :type id_: str
    :param customer_id: ID of the customer who placed the order
    :type customer_id: str
    :param status: Order lifecycle status
    :type status: OrderStatus
    :param total: Order total amount
    :type total: float
    :param currency: ISO 4217 currency code
    :type currency: str
    """

    id_: str = Field(
        alias="id", serialization_alias="id", description="Unique order identifier"
    )
    customer_id: str = Field(
        alias="customerId",
        serialization_alias="customerId",
        description="ID of the customer who placed the order",
    )
    status: OrderStatus = Field(description="Order lifecycle status")
    total: float = Field(description="Order total amount", ge=0)
    currency: str = Field(
        description="ISO 4217 currency code", min_length=3, max_length=3
    )

File: postman/sdks/Specs SDKs/Orders/cli/sdk/orders/order.go
package orders

import (
	"encoding/json"
	"example.com/orders/sdk/internal/unmarshal"
)

type Order struct {
	// Unique order identifier
	ID string `json:"id" xml:"id" required:"true"`
	// ID of the customer who placed the order
	CustomerID string `json:"customerId" xml:"customerId" required:"true"`
	// Order lifecycle status
	Status OrderStatus `json:"status" xml:"status" required:"true"`
	// Order total amount
	Total float64 `json:"total" xml:"total" required:"true" min:"0"`
	// ISO 4217 currency code
	Currency string `json:"currency" xml:"currency" required:"true" maxLength:"3" minLength:"3"`
}

func (o Order) String() string {
	jsonData, err := json.MarshalIndent(o, "", "  ")
	if err != nil {
		return "error converting struct: Order to string"
	}
	return string(jsonData)
}

func (o *Order) UnmarshalJSON(data []byte) error {
	if err := unmarshal.ValidateRequiredJSONKeys(data, o); err != nil {
		return err
	}
	type alias Order
	var tmp alias
	if err := json.Unmarshal(data, &tmp); err != nil {
		return err
	}
	*o = Order(tmp)
	return nil
}
